### PR TITLE
Arc-2494 redirect from get started

### PR DESCRIPTION
**Pull request by rachellerathbone**
**What's in this PR?**
Updated the manifest file to include useAsGetStarted and then updated the backend and frontend to utilise this new page and redirect from the get started page to the main admin page.

**Why**
We want to be able to render both the 'Get started' button + link in flag when someone installs the app, while continuing the app name render a link to the app in the left-hand panel. Forge doesn't support both and this was their workaround suggestion

**Affected issues**  
Arc-2494

**How has this been tested?**  
Locally and in staging.

e.g staging First part shows me refreshing while already on the main admin page (still works as expected). Then I demo from the manage apps screen: get started button is visible and when I click on it you can see get-started in the path (kind of... sorry I cut it off a little 😬 ) and then it redirects to the main admin page.

https://github.com/atlassian/jenkins-for-jira/assets/37155488/24bb6578-5a99-4619-9e62-4ea12dff3b4e

### CODE REVIEW

File: app/jenkins-for-jira-ui/src/api/redirectFromGetStarted.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const redirectFromGetStarted = async (): Promise<string> => {
+	const context: Context = await invoke('redirectFromGetStarted');
+	const {
+		siteUrl,
+		appId,
+		environmentId,
+		moduleKey
+	} = context;
+
+	if (moduleKey === 'get-started-page') {
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/api/redirectFromGetStarted.ts:19]: A complete URL for the admin page looks something like `<jirahost>/jira/settings/apps/get-started/<appId>/<environmentId>/`. There's no way to get get the values in between the host and the path (which here is just /) via the frontend (React/window) so I've had to pass it in from the request context defined in the backend (via Forge). This is needed to make sure we don't render any of the elements on the landing page when we hit get-started.

File: app/jenkins-for-jira-ui/src/components/JenkinsServerList/JenkinsServerList.tsx [hunk 2]
```diff
@@ -56,29 +65,44 @@ const JenkinsServerList = (): JSX.Element => {
 		</ButtonGroup>
 	);
 
-	return jenkinsServers?.length ? (
-		<>
-			<div className={headerContainer}>
-				<PageHeader actions={pageHeaderActions}>Jenkins configuration</PageHeader>
-			</div>
+	let contentToRender;
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/JenkinsServerList/JenkinsServerList.tsx:68]: eslint was bitching about nesting ternaries so had to extract the content we want to render so I we can handle all 3 scenarios.

File: app/jenkins-for-jira-ui/src/components/JenkinsServerList/JenkinsServerList.tsx [hunk 3]
```diff
@@ -56,29 +65,44 @@ const JenkinsServerList = (): JSX.Element => {
...
+				<ConnectedServers jenkinsServerList={jenkinsServers} refreshServers={fetchAllJenkinsServers} />
+			</>
+		);
+	} else if (moduleKey === 'get-started-page') {
+		contentToRender = (
+			<>
+				<div className={headerContainer}>
+					<PageHeader>Jenkins configuration</PageHeader>
+				</div>
+				<Spinner size='large' />
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/JenkinsServerList/JenkinsServerList.tsx:91]: Before the redirect kicks in, the empty/configured state was rendering briefly which is definitely not what we want. Accounted for that by rendering this spinner (matches what we see while waiting for the page to load).

File: app/manifest.yml [hunk 4]
```diff
@@ -7,35 +7,52 @@ modules:
...
       layout: basic
       icon: https://marketplace-cdn.atlassian.com/files/0c45cd96-ccf1-4490-9df4-307dded2f6e0?fileType=image&mode=full-fit
+
+    - key: get-started-page
+      resource: main
+      resolver:
+        function: redirectFromGetStarted
+      title: Jenkins for Jira
+      layout: basic
+      icon: https://marketplace-cdn.atlassian.com/files/0c45cd96-ccf1-4490-9df4-307dded2f6e0?fileType=image&mode=full-fit
```
Comment (joshkay10) [inline app/manifest.yml:17]: we re-use that icon a lot, might be time for

```
variables:
  iconUrl: https://marketplace-cdn.atlassian.com/files/0c45cd96-ccf1-4490-9df4-307dded2f6e0?fileType=image&mode=full-fit
  
...
modules:
...
      icon: ${{ variables.iconUrl }}
```

Comment (rachellerathbone) [reply app/manifest.yml:17]: Hahahaha you're funny. You remember this is a Forge app right where nothing is possible?

**Review by joshkay10 — APPROVED** [review]

**Pull request merged**